crates/partition-store/src/partition_store.rs

```diff
@@ -606,6 +609,33 @@ impl PartitionStore {
     pub fn partition(&self) -> &Arc<Partition> {
         self.db.partition()
     }
+
+    pub fn verify_deprecations(&self, key_range: RangeInclusive<PartitionKey>) {
```

Comment:
The key range should be available already in `self.partition_key_range()`, or do you have a scenario where partial range scan is needed?